middlewares/validators/events.js

```diff
@@ -165,6 +165,24 @@ const generateEventCode = async (req, res, next) => {
   }
 };
 
+const getEventCodes = async (req, res, next) => {
+  const { id } = req.params;
+
+  const schema = joi.object({
+    id: joi.string().required(),
+  });
+
+  const validationOptions = { abortEarly: false };
+
+  try {
+    await schema.validateAsync({ id }, validationOptions);
```

Comment:
next() function is called before the res.boom.badRequest() function. This means that the res.boom.badRequest() function will not be executed if the validation fails.

Reply:
but the validateAsync function will throw error if validation fails resulting in going to catch block